### DISCUSSION TITLE
The ability to toggle if other players can clone your vehicle.

### DISCUSSION THREAD

**Discussion by TheNameisZero** [Starystars67/BeamNG-MP#295] [Ideas]
Anyone who plays beamng and I mean actually plays the game. Will know that making a car that drives well, looks stunning, or is just for a good time isn't easy. We all remember that time we saw the vehicle option menu it was confusing and annoying to navigate. Especially if you are unfamiliar. But for the people who have learned it and swapped out every piece of their ride trying to get that engine to crank out a few more HPs, it's not simple. anyone can drag a boost gauge to full. But the guys tuning the suspension are the ones taking the time to get it just right. Ripping on turns in a station wagon still able to see that sports car in front you. But then it hits you like the guest account who just spawned in a rocket bus. What's that? The thing that pulled up right beside you? Woah! That looks like, no can't be. My car? *gasps* you might say "it can't be!" oh but it is my friend. That car you just built/tuned for 20 minutes has just been picked up with the push of a button. Ain't that a bummer? A real vibe killer if you will? Absolutely. All I'm saying is the people who boot up the game and just select a stock car or copy someone else are missing out on a large portion of the game.  I would like all players to have the option to toggle if their vehicle can be cloned. The game wouldn't let me save cars in custom variants if we were all supposed to be driving the same stuff.

**Comment by OfficialLambdax** [comment]
In order for the other cars to be fully shown on your client or in order for your car to be fully shown on other clients, you and they have to receive the full vehicle config. And with that you and they have the config. Removing or disabling the cloning/save/set default button wouldnt resolve the issue as everyone still has the config of everyone else. Doing that change would only keep the less the techy savy from stealing configs, however only for aslong until someone writes and publishs a mod that reenables it.

Another possible solution would be to not broadcast the whole vehicle config, but only the visible part. So the parts, but not the variables (the tuning settings). That however wouldnt work with all vehicles. Imagine you build a rally car with a heightend suspension. Now you spawn the car in multiplayer, but since the vars arent transmitted, the vehicle will not have the hightend suspension on the screens of the other players. So while you can neatly drive over rocks, on the screens of the others it will look like that you crash into every rock.

What im trying to say is, that i dont see any possible solution to combat config stealing without sacraficing the multiplayer experience

**Reply by 1eob** [reply]
The only thing i can think of is the ability to disable switching to the persons car

**Reply by TheNameisZero** [reply]
Thanks for laying out the obstacles, i think you have the solution right here. "Removing or disabling the cloning/save/set default button wouldnt resolve the issue as everyone still has the config of everyone else. Doing that change would only keep the less the techy savy from stealing configs, however only for aslong until someone writes and publishs a mod that reenables it." This works just fine and seems like the easiest answer.  Of course someones going to make a mod that reverts a change they don't like always do. Would there be a way to make it work like this? Once a player is spectating another player and cloned their asset the game will just auto delete it the way they enforce one vehicle per person or an "illegal vehicle"? Or like you suggested just hiding the button if the other player has the option toggled?  Super informative, thank you.

**Reply by OfficialLambdax** [reply]
No client side script can remove a users ability to get their hand on anothers vehicle. The mentioned option would just make it harder, but not really. As said a mod will make it useless, even a wireshark capture will make it useless. The only way to prevent full config stealing would be to not transmit the whole config, but then again that will render the player vehicles unstable. Another example would be of non abs vehicles. Someone might set the brake multiplier down so that he can fully press the brakes without locking up. Well, but for all other players he will lockup and will smoke a hell lot whenever he brakes

**Reply by OfficialLambdax** [reply]
Players even already figured how to bypass the mod loading denial in a server and they share that knowledge with others. Now the way that it works tho makes the game less and less stable the more mods the player tries to get loaded. And well those players then come to the dc of beammp, beamng, the forums, reddit etc. and complain that their game doesnt work, they cant join x server or that they cant deactivate their mods. What im trying to say is, that such a system, like the mod suppression feature is easiely bypassable. So easiely actually that it isnt even worth considering the option where we block certain buttons.

**Reply by TheNameisZero** [reply]
First days of mod loading denial was very nice. After reading your response i have to agree that it would not be a productive use of time. Anti cheat is one of the HARDEST things to do in any multiplayer game. Even battlefield hands it out to other companies. Thanks for your responses.
